lighthouse-cli/performance-experiment/report/scripts/perf-x.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* global window, document, location, fetch */
+
+'use strict';
+
+/**
+ * @fileoverview Report script for Project Performance Experiment.
+ *
+ * Include functions for supporting interation between report page and Perf-X server.
+ */
+
+window.addEventListener('DOMContentLoaded', () => {
+  new ConfigPanel();
+});
+
+class ConfigPanel {
+  constructor() {
+    this._configPanel = document.querySelector('.js-config-panel');
+    this._messageField = this._configPanel.querySelector('.js-message');
+    this._urlBlockingList = this._configPanel.querySelector('.js-url-blocking-patterns');
+    this._urlBlockingStatus = {};
+
+    const bodyToggle = this._configPanel.querySelector('.js-panel-toggle');
+    bodyToggle.addEventListener('click', this._toggleBody.bind(this));
+
+    const rerunButton = this._configPanel.querySelector('.js-rerun-button');
+    rerunButton.addEventListener('click', this._rerunLighthouse.bind(this));
+
+    // init list view buttons
+    const addButton = this._urlBlockingList.querySelector('.js-add-button');
+    const patternInput = this._urlBlockingList.querySelector('.js-pattern-input');
+    addButton.addEventListener('click', () => {
+      if (patternInput.value) {
+        this.addBlockedUrlPattern(patternInput.value);
+        this._urlBlockingList.parentNode.scrollTop = this._urlBlockingList.offsetHeight;
+        patternInput.value = '';
+      }
+    });
+    patternInput.addEventListener('keypress', event => {
+      (event.keyCode || event.which) === 13 && addButton.click();
+    });
+
+    // init tree view buttons
+    const requestBlockToggles = this._configPanel.querySelectorAll('.js-request-blocking-toggle');
+    requestBlockToggles.forEach(toggle => {
```

Comment:
Yea `[...requestBlockToggles]` is creating an array from the NodeList. You could also do `Array.from(requestBlockToggles)`. Both are more modern ways to `Array.prototype.forEach.call`. I would prefer the newer options over the latter. Perf will be negligible, esp since the list won't be that long.